src/sqlparser/src/parser.rs

```diff
@@ -1361,6 +1365,7 @@ impl Parser {
             Token::Pipe => Ok(21),
             Token::Caret | Token::Sharp | Token::ShiftRight | Token::ShiftLeft => Ok(22),
             Token::Ampersand => Ok(23),
+            Token::Arrow | Token::LongArrow | Token::HashArrow | Token::HashLongArrow => Ok(25),
```

Comment:
These precedence magic numbers will be refactored to fully align with PostgreSQL as part of #8140. For now it just needs to be greater than `BETWEEN` (21) and smaller than `+` (30).